addons/mail/models/mail_render_mixin.py

```diff
@@ -219,6 +236,46 @@ def _prepend_preview(self, html, preview):
             """).format(preview_markup)
             return tools.prepend_html_content(html, html_preview)
         return html
+
+    # ------------------------------------------------------------
+    # SECURITY
+    # ------------------------------------------------------------
+
+    def _is_dynamic(self):
+        for template in self:
+            for fname, field in template._fields.items():
+                engine = getattr(field, 'render_engine', 'inline_template')
+                if engine in ('qweb', 'qweb_view'):
+                    if self._is_dynamic_template_qweb(template[fname]):
+                        return True
+                else:
+                    if self._is_dynamic_template_inline_template(template[fname]):
```

Comment:
To handle other rendering engines
```suggestion
                    func = getattr(self, f"_is_dynamic_template_{engine}")
                    if func(template[fname]):
```